modules/calib/include/opencv2/calib.hpp

```diff
@@ -1122,6 +1122,40 @@ CV_EXPORTS_W double stereoCalibrate( InputArrayOfArrays objectPoints,
                                      OutputArray perViewErrors, int flags = CALIB_FIX_INTRINSIC,
                                      TermCriteria criteria = TermCriteria(TermCriteria::COUNT+TermCriteria::EPS, 30, 1e-6) );
 
+/** @brief Estimates intrinsics and extrinsics (camera pose) for multi-camera system.
+Multiview calibraton.
+
+@param[in] objPoints Calibration pattern object points. Expected shape: NUM_FRAMES x NUM_POINTS x 3. Supported data type: CV_32F.
+@param[in] imagePoints Detected pattern points on camera images. Expected shape: NUM_CAMERAS x NUM_FRAMES x NUM_POINTS x 2.
+@param[in] imageSize   Images resolution.
+@param[in] detectionMask - Pattern detection mask. Each value defines if i-camera observes calibration pattern in j moment of time. Expected size: NUM_CAMERAS x NUM_FRAMES. Expected type: CV_8U.
+@param[in] isFisheye indicates whether i-th camera is fisheye. In case if the input data contains mix of pinhole and fisheye cameras Rational distortion model is used. See @ref CALIB_RATIONAL_MODEL for details. Expected type: CV_8U.
```

Comment:
fisheye has different `CALIB_` constant (which are used for flagsForIntrinsics).

`CALIB_FIX_K4 = 1 << 7`

vs 

`CALIB_FIX_K4 = 0x00800`

So we can't properly handle mess of these models.
Or we should have `flagsForIntrinsics` on per-camera basis.